### DISCUSSION TITLE
Generated C++ can't construct in a single line

### DISCUSSION THREAD

**Discussion by tmayoff** [eProsima/Fast-DDS-Gen#233] [Ideas]
This might be related to this issue, https://github.com/eProsima/Fast-DDS-Gen/issues/38. For testing I like to setup my tests like this: 
```C++
const std::vector<Testing::TestMessage> tests = {
      {uuid, 0, "testing", 0},
      {uuid, 1, "Test Message", 1414241},
      {uuid, 2, "Messaging Test", 1245265346},
      {uuid, 3, "Test", std::chrono::system_clock::now().time_since_epoch().count()},
  };
```

However due to the way the constructors are generated I can't, is it possible to update the constructors to allow this (maybe with some of the suggestion in the linked issue)

**Comment by tmayoff** [comment]
The same would be nice with the QoS policies. Although I'm not sure if these are generated from IDL
```C++
fastdds::DurabilityServiceQosPolicy{Duration_t{}, fastdds::KEEP_ALL_HISTORY_QOS, 1, 10, 10, 10}},
```

**Comment by JLBuenoLopez** [comment]
Moving the ticket to the Ideas discussion forum where enhancements and feature requests are discussed.
